lib/python-indent.coffee

```diff
@@ -42,31 +47,29 @@ module.exports = PythonIndent =
 
     # Get base variables
     row = editor.getCursorBufferPosition().row
-    previousLine = editor.buffer.lineForRow(row - 1)
-    return unless row
-
-    # Based on configuration, branch to corresponding logic
-    if (matchOpeningDelimiter = PythonIndent.openingDelimiterIndentRegex.exec previousLine) isnt null
-      PythonIndent.indentOnOpeningDelimiter(editor, row, previousLine, matchOpeningDelimiter)
-    else if PythonIndent.hangingIndentRegex.test previousLine
-      PythonIndent.indentHanging(editor, row, previousLine)
-    else if PythonIndent.openingDelimiterUnindentRegex.test previousLine
-      PythonIndent.unindentOnOpeningDelimiter(editor, row, previousLine)
-
-  indentOnOpeningDelimiter: (editor, row, previousLine, matchOpeningDelimiter) ->
-    # Get index of opening delimiter.
-    # The indent should be one whitespace character past this (sorry, no tabs)
-    indentColumn = previousLine.lastIndexOf(matchOpeningDelimiter[1]) + 1
+    col = editor.getCursorBufferPosition().column
+
+    # Parse the entire file up to the current point, keeping track of brackets
+    lines = editor.getTextInBufferRange([[0,0], [row, col]]).split('\n')
```

Comment:
I do think of potential performance issues on very large (> 100 lines) files. I think we'll want to put some load with lots of brackets to test potentially parsing the whole file.